client/reader.js

```diff
@@ -74,18 +74,26 @@ rl.on('line', (line) => {
     }
   } else if (line.indexOf('FLEX: ') > -1) {
     address = line.match(/FLEX:.*?\[(\d*?)\] /)[1].trim();
-    if (line.match( /( ALN | GPN | NUM)/ )) {
-      message = line.match(/FLEX:.*?\[\d*\] ... (.*?)$/)[1].trim();
-      trimMessage = message;
-    } else {
-      message = false;
-      trimMessage = '';
+      if (line.match( /1600\/2\/F\/A/ )) {
+       if (line.match( /( ALN | GPN | NUM)/ )) {
```

Comment:
This if is repeated in each of the three `/1600\/2\/F\/A/` ifs, I'd suggest moving it one level up instead, e.g.

```
if (line.match( /( ALN | GPN | NUM)/ )) {
  if (line.match( /1600\/2\/F\/A/ )) {
    frag = line.match(/FLEX:.*?\[\d*\] ... (.*?)$/)[1].trim();
    message = false;
    trimMessage = '';
  }
  else if (line.match( /1600\/2\/C\/A/ )) {
  ...
  }
  ...
}
```